### ISSUE TITLE
Failure to serialize an array of tables.

### ISSUE THREAD

**Issue by alexchandel** [karupanerura/toml#3]
The following seemingly trivial code fails on OS X, perl5.18, TOML 0.96, TOML::Parser 0.5:
```perl
use strict;
use warnings;
use feature qw(say);
use TOML;

say to_toml {
    suites => [
        {
            name => "msvc",
        }
    ]
};
```

It fails with the message `Bad type in to_toml: HASH at -e line 7.`

Deserializing the equivalent TOML succeeds, but reserializing that fails.

**Comment by karupanerura** [comment]
Thank you for noticing this issue.
Can you try to use TOML::Dumper?
This is my new TOML dumper module. (not yet uploaded to CPAN)
If this module can solve your problem, I'll integrate TOML::Dumper to TOML.pm.
https://github.com/karupanerura/TOML-Dumper#synopsis

**Comment by cxw42** [comment]
@karupanerura Yes, would you please integrate TOML::Dumper into TOML?  For example, TOML::Dumper can dump blessed hashrefs, which TOML currently cannot (per [this check](https://metacpan.org/source/KARUPA/TOML-0.97/lib/TOML.pm#L33)).  Example:

    $ perl -MScalar::Util=reftype -MTOML -E 'package X; sub new { my $self = shift; return bless {foo=>42}; } package main; my $x=X->new; say join "\n", ref($x), reftype($x), to_toml(X->new)'
    You cannot convert non-HashRef values to TOML at -e line 1.
    $ perl -MScalar::Util=reftype -MTOML::Dumper -E 'package X; sub new { my $self = shift; return bless {foo=>42}; } package main; my $x=X->new; say join "\n", ref($x), reftype($x), TOML::Dumper->new->dump(X->new)'
    X
    HASH
    foo = 42

Thank you!

**Comment by karupanerura** [comment]
@cxw42 I see. OK.
But, I want to keep TOML's behavior for compatibility. So that error will be still occurred by default in the new version.
I'll consider providing a replacer interface like this: https://metacpan.org/pod/TOML#How-change-how-to-de-serialize?
Do you have any problem with this policy?

**Comment by cxw42** [comment]
@karupanerura Fine with me!  Thanks!
